src/main/features/claim/routes/pay.ts

```diff
@@ -85,27 +85,27 @@ export default express.Router()
         throw new Error('No amount entered, you cannot pay yet')
       }
 
-      const paymentId = draft.document.claimant.payment.id
+      const paymentRef = draft.document.claimant.payment ? draft.document.claimant.payment.reference : undefined
 
-      if (paymentId) {
+      if (paymentRef) {
         const payClient: PayClient = await getPayClient()
-        const payment: Payment = await payClient.retrieve(user, paymentId)
-        switch (payment.state.status) {
-          case 'success':
-            return res.redirect(Paths.finishPaymentReceiver.evaluateUri({ externalId: draft.document.externalId }))
+        const paymentResponse: PaymentRetrieveResponse = await payClient.retrieve(user, paymentRef)
+        if (paymentResponse !== undefined) {
```

Comment:
This is to ignore payment drafts from V1 which will result in new payment initiated in case previous payment was cancelled or declined.